### PR TITLE
🐛Fixes `clusterawsadm version` command

### LINKED ISSUE #1013: Release artifact clusterawsadm reports a missing or incorrect version
/kind bug

**What steps did you take and what happened:**
In v0.3.7, clusterawsadm reports no version:
```
$ curl --silent -output clusterawsadm -L https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v0.3.7/clusterawsadm-linux-amd64
$ chmod u+x clusterawsadm
$ ./clusterawsadm version
Version Info: GitReleaseTag: "{GIT_RELEASE_TAG}", MajorVersion: "{GIT_MAJOR}", MinorVersion:"{GIT_MINOR}", GitReleaseCommit:"{GIT_RELEASE_COMMIT}", GitTreeState:"{GIT_TREE_STATE}"
```

In v0.3.6, it reports version v0.3.5:
```
$ curl --silent -output clusterawsadm -L https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v0.3.6/clusterawsadm-linux-amd64
$ chmod u+x clusterawsadm
$ ./clusterawsadm version
Version Info: GitReleaseTag: "v0.3.5", MajorVersion: "0", MinorVersion:"3", GitReleaseCommit:"33e957884b5fa0", GitTreeState:"clean"
```

In v0.3.4, it reports the right version! :smile: 
```
$ curl --silent -output clusterawsadm -L https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v0.3.6/clusterawsadm-linux-amd64
$ chmod u+x clusterawsadm
$ ./clusterawsadm version
Version Info: GitReleaseTag: "v0.3.4", MajorVersion: "0", MinorVersion:"3", GitReleaseCommit:"3e1a814a362d93", GitTreeState:"clean"
```

**What did you expect to happen:**
`clustersawsadm version` should report the release version

**Anything else you would like to add:**

**Environment:**

- Cluster-api-provider-aws version: v0.3.7
- OS (e.g. from `/etc/os-release`):  Fedora 29

### PR DESCRIPTION
**Pull request by aaroniscode**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1013

### CODE REVIEW

**Comment by aaroniscode** [comment]
I'm interested in your thoughts on this approach. I took the `git` version environment variable code in `print-workspace-status.sh` and moved it into a the version::get_version_vars() function in `hack/version.sh`.  I modeled that file from k8s.io/hack/lib/version.sh.

I'm new to Go's `ldflags` and not familiar with the history of the `print-workspace-status.sh` script. Happy to change approach if necessary.

**Comment by randomvariable** [comment]
`print-workspace-status.sh` derivates from when Bazel was being used.

/ok-to-test

File: hack/print-workspace-status.sh [hunk 1]
```diff
@@ -17,58 +17,10 @@ set -o errexit
 set -o nounset
```
Comment (ncdc) [inline hack/print-workspace-status.sh]: Can't we delete this file?

Comment (randomvariable) [reply hack/print-workspace-status.sh]: Actually, yes, I think so.

Comment (aaroniscode) [reply hack/print-workspace-status.sh]: @ncdc I left the `print-workspace-status.sh` file as I wasn't sure what it was used for. But happy to delete it if it's not needed any more.

Comment (ncdc) [reply hack/print-workspace-status.sh]: Delete away!

Comment (aaroniscode) [reply hack/print-workspace-status.sh]: deleted!

**Pull request merged**